### DISCUSSION TITLE
WASM - "Incremental output" or "split output"

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#18955] [Q&A]
The question might be weird, but: is there a way or do you see the possibility to build a Uno-WASM-app where there is some "static" output and some "dynamic" one?

My context: I build my app using a github action and then publish it to Storj DCS, a decentralized cloud storage. Storj allows me to [host a website from static files](https://storj.dev/dcs/code/static-site-hosting). This is awesom! No need to run a server by myself, just upload the files and after initial setup (adding some DNS-entries) my uno app can be accessed from everywhere.

But if I do an update to my app and deploy again, it creates a different package_GUID-folder containing the assets and all clr-types and what not. Most of the stuff in there is really static, like fonts, splash screen and such. The only thing that changed most of the time is one or two dlls of mine. But I always need to upload everything again and next to the existing one. I also have to take care to not upload the new index.html prior to the package_GUID-folder as the app would otherwise not work until all assets are uploaded.

So the question is: can I create a more static build where existing files are kept or something similar?

**Comment by jeromelaban** [comment]
This behavior has changed, somehow, in net9 where the DLLs are individually named with a hash. The problem is the other assets that need to be individually hashed and referenced properly without know what the hash will be. We'll likely find something to help, but this is not entirely handled by the .NET SDK at this time for blazor either.

For net9, you have less to upload, but you'll still have to upload everything anyways.

**Reply by TopperDEL** [reply]
Thanks! So switching to net9 might help a bit, at least regarding the amount to upload?

**Reply by jeromelaban** [reply]
That's correct, unless you have very large assets in which case net9 won't change much.
